hl7/4.2/modules/ROOT/edits/hl7-connector-config-topics.adoc

```diff
@@ -0,0 +1,269 @@
+
+[source,yaml,linenums]
+----
+form: HL7
+version: '2.5.1'
+imports: [ '/hl7/v2_5_1/basedefs.esl' ]
+id: 'ADT_A05'
+name: 'ADT_A05'
+data:
+- { idRef: 'MSH', usage: M }
+- { idRef: 'SFT', usage: O, count: '>1' }
+- { idRef: 'EVN', usage: M }
+- { idRef: 'PID', usage: M }
+- { idRef: 'PD1', usage: O }
+- { idRef: 'CON', usage: O, count: '>1' }
+- { idRef: 'NTE', usage: O }
+- { idRef: 'ROL', usage: O, count: '>1' }
+- { idRef: 'NK1', usage: O, count: '>1' }
+- { idRef: 'PV1', usage: M }
+- { idRef: 'PV2', usage: O }
+- { idRef: 'CON', usage: O, count: '>1' }
+- { idRef: 'ROL', usage: O, count: '>1' }
+...
+----
+
+If you want to define a non-standard segment for your message structure,
+add the segment definition to the schema. This is more complex than just
+modifying the segment structure, as you need to list all components
+in the segment. The easiest starting point for this is to find a similar
+standard HL7 segment and copy the definition used for that standard
+segment from the `basedefs.esl` file. You can then add the segments key
+following your message structure definition, followed by one or more
+segment definitions.
+
+=== Set Your Schema Locations
+
+You can configure schema locations in the Anypoint Studio XML view.
+
+. Switch to the XML view by clicking *Configuration XML*.
+
+. Modify your HL7 EDI configuration to include a list of all the schemas to include by adding an `+<http://edischema[edi:schema]>+` element for each document type:
++
+[source,xml,linenums]
+----
+<hl7-edi:config name="HL7_EDI__Configuration" identKeys="true" doc:name="HL7 EDI: Configuration">
+  <hl7-edi:schemas>
+    <hl7-edi:schema>hl7/v2_6/ADT_A01.esl</hl7-edi:schema>
+  </hl7-edi:schemas>
+</hl7-edi:config>
+----
++
+. After you create a global element for your HL7 EDI, configure the message structure, operations, and acknowledgments.
+
+== Event and Message to Message Structure Map
+
+If you configure the connector with multiple message structure schemas
+(whether in separate schema definitions files, as with the provided schemas,
+or in a single file) you need to define a mapping from the HL7 event
+and message types to message structures.
+
+HL7 defines the Message Type in the component values of `MSH-09`, a composite
+structure of type MSG. HL7 EDI Connector uses these component values to find the structure schema to use for processing a receive message, according to the following rules:
+
+* If the MSG-01 Message Code value is `ACK`, always use the predefined ACK schema.
+* If the MSG-03 Message Structure value is present (a value such
+as `ADT_A01`), use the schema structure with that ID.
+* If neither of these, then use a configured Event and Message to Message Structure Map to determine
+the message structure from the specified Trigger Event (`MSG-02`) and Message Code (`MSG-01`) values.
+
+The Event and Message to Message Structure Map is an optional configuration parameter.
```

Comment:
GLOBAL: hyphenate message-to-message if it is a compound adjective, as I suspect.